src/backend/core/api/decorators.py

```diff
@@ -0,0 +1,50 @@
+"""Decorators for Impress' core application."""
+
+import os
+from ipaddress import ip_address, ip_network
+
+from django.http import HttpResponseForbidden
+
+
+def monitoring_cidr_protected_view(view):
+    """
+    Decorator to restrict access to a view based on CIDR ranges.
+
+    Checks the client's IP address against allowed CIDR ranges specified
+    in the MONITORING_ALLOWED_CIDR_RANGES environment variable. If the
+    IP address is not within the allowed ranges, access is denied.
+    """
+
+    def wrapped_view(request, *args, **kwargs):
+        cidr_env_raw = os.environ.get("MONITORING_ALLOWED_CIDR_RANGES", "")
+        cidr_env_stripped = cidr_env_raw.strip().strip('"').strip("'")
+
+        allow_all = cidr_env_stripped == "*"
+
+        allowed_cidr_ranges = []
+        if not allow_all:
+            try:
+                allowed_cidr_ranges = [
+                    ip_network(c.strip().strip('"').strip("'"))
+                    for c in cidr_env_stripped.split(",")
+                    if c.strip()
+                ]
+            except ValueError as e:
+                raise ValueError(
+                    f"Invalid CIDR range in MONITORING_ALLOWED_CIDR_RANGES: {e}"
+                ) from e
+
+        client_ip = request.META.get("REMOTE_ADDR")
```

Comment:
We can't rely on this security. First this value will probably wrong if the django application is behind multiple proxy. Also it is possible to spoff the ip.

I look how to secure a prometheus metrics endpoint and I found that it can be secure using a basic auth mechanisms. DRF provides a BaseAuth backend : https://www.django-rest-framework.org/api-guide/authentication/#basicauthentication

Reply:
I removed CIDR protection and switched to a BasicAuth implementation.
Looks like DRF needs a user for BaseAuth, so i created a dummy user `MonitoringUser`.
Indeed its way better to like this 👍🏼